processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -328,6 +329,12 @@ func (c *WatchClient) extractPodAttributes(pod *api_v1.Pod) map[string]string {
 					tags[conventions.AttributeK8SStatefulSetName] = ref.Name
 				}
 			case "Job":
+				if c.Rules.CronJobName {
+					parts := c.deploymentRegex.FindStringSubmatch(ref.Name)
```

Comment:
I don't think `deploymentRegex` can be applied to job names, e.g. `auth-cronjob-27667920` job would give `auth` as a cronjob name